patches/api/0008-Paper-Plugins.patch

```diff
@@ -0,0 +1,2486 @@
++     * @param onError   a runnable that is executed before any unknown exception is raised through a sneaky throw.
++     * @param <T>       the generic type of the parent class the created object will be cast to
++     * @return the object instantiated from the class found at the provided fully qualified class name, cast to the
++     * parent type
++     */
++    @NotNull
++    public static <T> T loadClass(@NotNull String clazz, @NotNull Class<T> classType, @NotNull ClassLoader loader, @Nullable Runnable onError) {
++        try {
++            T clazzInstance;
++
++            try {
++                Class<?> jarClass = Class.forName(clazz, true, loader);
++
++                Class<? extends T> pluginClass;
++                try {
++                    pluginClass = jarClass.asSubclass(classType);
++                } catch (ClassCastException ex) {
++                    throw new ClassCastException("class '%s' does not extend '%s'".formatted(clazz, classType));
++                }
++
++                clazzInstance = pluginClass.getDeclaredConstructor().newInstance();
++            } catch (IllegalAccessException exception) {
++                throw new RuntimeException("No public constructor");
++            } catch (InstantiationException exception) {
++                throw new RuntimeException("Abnormal class instantiation", exception);
++            }
++
++            return clazzInstance;
++        } catch (Throwable e) {
++            if (onError != null) {
++                onError.run();
++            }
++            SneakyThrow.sneaky(e);
++        }
++
++        throw new AssertionError(); // Shouldn't happen
++    }
++
++}
+diff --git a/src/main/java/io/papermc/paper/registry/RegistryAccess.java b/src/main/java/io/papermc/paper/registry/RegistryAccess.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..ce66c664003e80087df88bded37ef17bd398bf85
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/registry/RegistryAccess.java
+@@ -0,0 +1,16 @@
++package io.papermc.paper.registry;
++
++import org.jetbrains.annotations.NotNull;
++
++public class RegistryAccess {
++
++    /**
++     * Stub
++     *
++     * @return
++     */
++    @NotNull
++    public static RegistryAccess getInstance() {
++        return new RegistryAccess();
++    }
++}
+diff --git a/src/main/java/org/bukkit/command/PluginCommand.java b/src/main/java/org/bukkit/command/PluginCommand.java
+index 1dbbc244309043b18c1d71707c4fb066c0d0e02d..08d3ff24d5318b0c77d6467bcb705bffac202842 100644
+--- a/src/main/java/org/bukkit/command/PluginCommand.java
++++ b/src/main/java/org/bukkit/command/PluginCommand.java
+@@ -14,7 +14,7 @@ public final class PluginCommand extends Command implements PluginIdentifiableCo
+     private CommandExecutor executor;
+     private TabCompleter completer;
+ 
+-    protected PluginCommand(@NotNull String name, @NotNull Plugin owner) {
++    public PluginCommand(@NotNull String name, @NotNull Plugin owner) { // Paper
```

Comment:
Why is this now public?

Reply:
Reverted that. This was when paper plugins were able to register their own plugin commands.